webpack/assets/javascripts/react_app/components/SettingUpdateModal/components/SettingForm/SettingForm.js

```diff
@@ -14,32 +13,33 @@ const SettingForm = ({
   setModalClosed,
   submitForm,
 }) => {
-  const handleSubmit = async (values, actions) => {
+  const handleSubmit = values => {
     let submitValues = { setting: values };
 
     if (setting && setting.settingsType === 'array') {
       const splitValue = values.value === '' ? [] : values.value.split(',');
       submitValues = { setting: { value: splitValue } };
     }
 
-    await submitForm({
+    return submitForm({
       url: SETTING_UPDATE_PATH.replace(':id', setting.id),
       values: submitValues,
       item: 'Settings',
       message: __('Setting was successfully updated.'),
       method: 'put',
     });
-    setModalClosed();
   };
 
   return (
     <ForemanForm
+      onSuccess={setModalClosed}
       onSubmit={(values, actions) => handleSubmit(values, actions)}
       initialValues={initialValues}
       onCancel={setModalClosed}
     >
-      <FormikField
+      <TextField
```

Comment:
Not all settings are displayed in text field, selections and boolean display in dropdown.